smithy-model/src/main/java/software/amazon/smithy/model/traits/DeprecatedTrait.java

```diff
@@ -72,6 +74,17 @@ public Optional<String> getMessage() {
         return Optional.ofNullable(message);
     }
 
+    public String getDeprecatedDescription(ShapeType shapeType) {
+        String shapeTypeString = shapeType == ShapeType.OPERATION ? "operation" : "shape";
+        if (Objects.isNull(this.since)) {
+            return this.message != null ? this.message : "This " + shapeTypeString + " is deprecated.";
+        } else if (Objects.isNull(this.message)) {
+            return "This " + shapeTypeString + " is deprecated since " + this.since;
+        } else {
+            return "This " + shapeTypeString + " is deprecated since " + this.since + ": " + this.message;
+        }
```

Comment:
This would be less duplicate using a `StringBuilder`:

```suggestion
        StringBuilder builder = new StringBuilder("This ");
        builder.append(shapeType == ShapeType.OPERATION ? "operation" : "shape").append(" is deprecated");
        if (since != null) {
            builder.append(" since ").append(since);
        }
        if (message != null) {
            builder.append(": ").append(message);
        } else {
            builder.append(".");
        }
        return builder.toString();
```